src/api/expense.controller.js

```diff
@@ -0,0 +1,94 @@
+const { expenseService } = require('../services/expense.service.js');
+const { userService } = require('../services/user.service.js');
+
+const getExpenses = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
```

Comment:
Consider validating the `from` and `to` query parameters to ensure they are in the correct date format before passing them to the `expenseService.getExpenses` function. This can prevent potential errors or unexpected behavior if invalid dates are provided.